File: mergeSort.py
def merge(low,mid,high,arr):
    left = []
    for i in range(low,mid+1):
        left.append(arr[i])
    
    right = []
    for i in range(mid+1,high+1):
        right.append(arr[i])
    
    lIndex,rIndex = 0,0
    arrIndex = low
    while lIndex < len(left) and rIndex < len(right):
        if(left[lIndex]<right[rIndex]):
            arr[arrIndex] = left[lIndex]
            lIndex+=1
            arrIndex+=1
        else:
            arr[arrIndex] = right[rIndex]
            rIndex+=1
            arrIndex+=1
    
    while lIndex < len(left):
        arr[arrIndex] = left[lIndex]
        arrIndex+=1
        lIndex+=1
    
    while rIndex < len(right):
        arr[arrIndex] = right[rIndex]
        rIndex+=1
        arrIndex+=1
    #print(arr)
    

def mergeSort(low,high,arr):
    if(low<high):
        mid = (low+high)//2
        mergeSort(low,mid,arr)
        mergeSort(mid+1,high,arr)
        merge(low,mid,high,arr)

def Merge_Sort():
    arr = [ 2, 3,4, 1, 5, 6,23,21,18,8,17]
    #merge(0,2,4,arr)
    mergeSort(0,len(arr)-1,arr)
    print(arr)
Merge_Sort()
"""arr = [ 2, 3, 4,17, 1, 5, 6,14]
merge(0,3,len(arr)-1,arr)
print(arr)"""

File: MergeSort.java
class MergeSort {
    public void merge(int arr[],int low ,int mid,int high){
        int n1 = mid-low+1;
        int n2 = high-mid;

        int L[] = new int[n1];
        int R[] = new int[n2];
    
        for(int k=0;k<n1;k++)
            L[k] = arr[low+k];
        for(int l=0;l<n2;l++)
            R[l] = arr[mid+1+l];

        int i = 0,j=0;
        int index = low;
        while(i<n1 && j<n2){
            if(L[i]<=R[j]){
                arr[index++] = L[i++];
            }
            else{
                arr[index++] = R[j++];
            }
        }

        while(i<n1){
            arr[index++] = L[i++];
        }

        while(j<n2){
            arr[index++] = R[j++];
        }
    }

    public void mergeSort(int arr[],int low,int high){
        if(low==high){
            return ;
        }
        int mid = (low+high)/2;
        mergeSort(arr,low,mid);
        mergeSort(arr,mid+1,high);
        merge(arr,low,mid,high);
    }
    public static void main(String[] arg){
        MergeSort obj = new MergeSort();

        
        int arr[] = {10,12,19,5,6,8,5,4,3,2,14, 34, 91, 68, 69, 1, 89, 50, 4, 64, 54, 87, 53, 76, 58, 37, 15, 61, 29, 51, 57, 95, 100, 40, 29, 1, 48, 60, 36};
        int low = 0,high = arr.length-1;
        obj.mergeSort(arr,low,high);

        for(int i=0;i<high+1;i++){
            System.out.println(arr[i]);
        }
    }    
}
